.github/actions/run-tests/action.yml

```diff
@@ -0,0 +1,127 @@
+name: 'ZKVyper Tests'
+description: 'Executes zkvyper unit and integration tests'
+inputs:
+  target:
+    description: 'Specific target triplet.'
+    required: false
+    default: ''
+  download-zkvyper:
+    description: 'Whether to download prebuilt zkvyper executable (set to `true` if testing prebuilt binaries).'
+    required: false
+    default: 'false'
+  results-xml:
+    description: 'Output unit tests results XML filename.'
+    required: false
+    default: 'unit-tests-results.xml'
+  zkvyper-version:
+    description: 'Prebuilt version of zkvyper compiler to download and use in tests.'
+    required: false
+    default: '0.3.10'
+  vyper-version:
+    description: 'Version of Vyper compiler to download.'
+    required: false
+    default: '0.3.10'
+  vyper-commit:
+    description: 'Commit of Vyper compiler to download.'
+    required: false
+    default: '91361694'
+runs:
+  using: "composite"
+  steps:
+    - name: Install vyper compiler
+      shell: ${{ runner.os == 'Windows' && 'msys2 {0}' || 'bash' }}
+      env:
+        VYPER_DOWNLOAD_URL: "https://github.com/vyperlang/vyper/releases/download"
+        VYPER_VERSION: ${{ inputs.vyper-version || '0.3.10' }}
```

Comment:
Unfortunately, we must duplicate default values here, otherwise the correct parameter passing won't work. The design flaw in GHA.